tfsdklog/sink.go

```diff
@@ -0,0 +1,96 @@
+package tfsdklog
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/terraform-plugin-log/internal/logging"
+)
+
+const (
+	envLog     = "TF_LOG"
+	envLogFile = "TF_LOG_PATH"
+)
+
+var sink hclog.Logger
+
+func init() {
+	sink = newSink()
+}
+
+// ValidLevels are the string representations of levels that can be set for
+// loggers.
+var ValidLevels = []string{"TRACE", "DEBUG", "INFO", "WARN", "ERROR", "OFF"}
```

Comment:
Nit: Might be good to expose these levels (and `"JSON"`) as constants.

Reply:
I think this is the only place those strings are used. :/

Reply:
True, we can always add later, if necessary. Just a habit I got into so magic strings can be individually documented.